src/components/Common/CrudlDocument.vue

```diff
@@ -117,141 +111,107 @@
-      collection: String,
-      documents: Array,
-      displayBulkDelete: Boolean,
-      displayCreate: {
-        type: Boolean,
-        default: false
-      },
-      allChecked: Boolean,
-      totalDocuments: Number,
-      lengthDocument: {
-        type: Number,
-        default: 0
-      },
-      selectedDocuments: Array,
-      paginationFrom: Number,
-      paginationSize: Number,
-      searchTerm: String,
-      rawFilter: Object,
-      basicFilter: Array,
-      sorting: Object,
-      availableFilters: Object,
-      documentToDelete: String,
-      performDelete: Function
+    allChecked: Boolean,
+    totalDocuments: Number,
+    lengthDocument: {
+      type: Number,
+      default: 0
     },
-    data () {
-      return {
-        formatFromBasicSearch,
-        formatSort,
-        documentIdToDelete: '',
-        singleDeleteIsOpen: false,
-        bulkDeleteIsOpen: false,
-        isLoading: false
-      }
+    selectedDocuments: Array,
+    paginationFrom: Number,
+    paginationSize: Number,
+    currentFilter: Object,
+    sorting: Object,
+    searchFilterOperands: Object,
+    documentToDelete: String,
+    performDelete: Function
+  },
+  data() {
+    return {
+      documentIdToDelete: '',
+      singleDeleteIsOpen: false,
+      bulkDeleteIsOpen: false,
+      isLoading: false
+    }
+  },
+  methods: {
+    onCreateClicked() {
+      this.$emit('create-clicked')
     },
-    methods: {
-      create () {
-        this.$emit('create-clicked')
-      },
-      changePage (from) {
-        this.$router.push({query: {...this.$route.query, from}})
-      },
-      confirmBulkDelete () {
-        this.isLoading = true
-        this.performDelete(this.index, this.collection, this.selectedDocuments)
-          .then(() => {
-            this.close()
-            this.refreshSearch()
-            this.isLoading = false
-            return null
-          })
-          .catch((e) => {
-            this.$store.commit(SET_TOAST, {text: e.message})
-          })
-      },
-      confirmSingleDelete (id) {
-        this.performDelete(this.index, this.collection, [id])
-          .then(() => {
-            this.close()
-            this.refreshSearch()
-            return null
-          })
-          .catch((e) => {
-            this.$store.commit(SET_TOAST, {text: e.message})
-          })
-      },
-      quickSearch (searchTerm) {
-        this.$router.push({query: {searchTerm, from: 0}}, () => {
-          this.$emit('crudl-refresh-search')
-        }, () => {
-          this.$emit('crudl-refresh-search')
+    changePage(from) {
+      this.onFiltersUpdated(
+        Object.assign(this.currentFilter, {
+          from
+        })
+      )
+    },
+    confirmBulkDelete() {
+      this.isLoading = true
+      this.performDelete(this.index, this.collection, this.selectedDocuments)
+        .then(() => {
+          this.close()
+          this.refreshSearch()
+          this.isLoading = false
+          return null
```

Comment:
Why returning null ?
(+ others below)

Reply:
I don't know. Even if the blame points on me, I didn't code these methods, I just formatted them via Prettier.